gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/AgentDirectoryMapper.java

```diff
@@ -0,0 +1,255 @@
+        humanName.getPrefix().add(getPractitionerPrefix(name.getPrefix()));
+        nameList.add(humanName);
+
+        return nameList;
+    }
+
+    private String getPractitionerFamily(String family) {
+        return StringUtils.isNotEmpty(family) ? family : UNKNOWN;
+    }
+
+    private StringType getPractitionerGiven(String given) {
+        return StringUtils.isNotEmpty(given) ? new StringType(given) : null;
+    }
+
+    private StringType getPractitionerPrefix(String prefix) {
+        return StringUtils.isNotEmpty(prefix) ? new StringType(prefix) : null;
+    }
+
+    private Organization createRepresentedOrganization(RCCTMT120101UK01Organization representedOrg, String id) {
+        var organization = new Organization();
+
+        organization
+            .setName(getOrganizationName(representedOrg.getName()))
+            .setId(id + ORG_ID_SUFFIX);
+        organization.getMeta().getProfile().add(new UriType(ORG_META_PROFILE));
+        organization.getIdentifier().add(getOrganizationIdentifier(representedOrg.getId()));
+        organization.getTelecom().add(getOrganizationTelecom(representedOrg.getTelecom()));
+        organization.getAddress().add(getOrganizationAddress(representedOrg.getAddr()));
+
+        return organization;
+    }
+
+    private Organization createAgentOrganization(RCCTMT120101UK01Organization agentOrg, String id, CV code) {
+        var organization = new Organization();
+        organization
+            .setName(getOrganizationName(agentOrg.getName()))
+            .setId(id);
+        organization.getMeta().getProfile().add(new UriType(ORG_META_PROFILE));
+        organization.getIdentifier().add(getOrganizationIdentifier(agentOrg.getId()));
+        organization.getType().add(getText(code));
+        organization.getTelecom().add(getOrganizationTelecom(agentOrg.getTelecom()));
+        organization.getAddress().add(getOrganizationAddress(agentOrg.getAddr()));
+
+        return organization;
+    }
+
+    private String getOrganizationName(String name) {
+        return name != null ? name : UNKNOWN;
+    }
+
+    private ContactPoint getOrganizationTelecom(List<TEL> telecomList) {
+        var telecom = telecomList.stream().findFirst();
+        if (telecom.isPresent()) {
+            var contactPoint = new ContactPoint();
+            return contactPoint
+                .setValue(getTelecomValue(telecom.get()))
+                .setSystem(ContactPoint.ContactPointSystem.PHONE)
+                .setUse(ContactPoint.ContactPointUse.WORK)
+                .setRank(TELECOM_RANK);
+        }
+
+        return null;
+    }
+
+    private String getTelecomValue(TEL telecom) {
+        var value = telecom.getValue();
+        if (value != null) {
+            return isWorkPlaceValue(telecom.getUse()) ? stripTelPrefix(value) : value;
+        }
+
+        return null;
+    }
+
+    private boolean isWorkPlaceValue(List<CsTelecommunicationAddressUse> use) {
```

Comment:
I think this method, `stripTelPrefix` and some address-related methods are in LocationMapper as well. Would it be worth to create a util class for those, which could hold all useful address & telecom methods?

Reply:
Great idea! I've created 2 seperate Util classes for these 😄